### PR TITLE
Add the freaking loads

### PR DESCRIPTION
**Pull request by ricfehr3**

### CODE REVIEW

File: src/cpu/instruction.rs [hunk 1]
```diff
@@ -70,6 +70,8 @@ pub enum Instruction {
     LdFromUpperMem(),
```
Comment (toastbucket) [inline src/cpu/instruction.rs:70]: Lets call these `LdToImmUpperMem` and `LdFromImmUpperMem`

File: src/cpu/instruction.rs [hunk 2]
```diff
@@ -70,6 +70,8 @@ pub enum Instruction {
...
     LdToImmMem(),
     LdFromImmMem(),
+    LdToCMem(),
```
Comment (toastbucket) [inline src/cpu/instruction.rs:73]: And these `LdToCUpperMem` and `LdFromCUpperMem`

File: src/cpu/test.rs [hunk 3]
```diff
@@ -1457,6 +1457,41 @@ fn test_ld_from_mem_imm() {
...
+#[test]
+fn test_ld_to_c_mem() {
+    let mut cpu = Cpu::new(Rc::new(RefCell::new(Mmu::new())));
+    const INSTRUCTIONS_LEN: usize = 1;
+    let test_ram: [u8; INSTRUCTIONS_LEN] = [
+        Instruction::LdToCMem().as_byte(),
+    ];
+
+    cpu.load_test_ram(&test_ram);
+    //cpu.write_byte(0xff69, 0x69);
```
Comment (toastbucket) [inline src/cpu/test.rs:1470]: remove

**Review by toastbucket — CHANGES_REQUESTED** [review]

File: src/cpu/test.rs [hunk 4]
```diff
@@ -1492,6 +1492,22 @@ fn test_ld_from_c_mem() {
...
+#[test]
+fn test_ld_sp_hl() {
+    let mut cpu = Cpu::new(Rc::new(RefCell::new(Mmu::new())));
+    const INSTRUCTIONS_LEN: usize = 1;
+    let test_ram: [u8; INSTRUCTIONS_LEN] = [
+        Instruction::LdHlToSp().as_byte(),
+    ];
+
+    cpu.load_test_ram(&test_ram);
+    cpu.set_reg_16(Register16Bit::HL, 0x69);
```
Comment (toastbucket) [inline src/cpu/test.rs:1505]: Lets test with a `u16` since these are 16 bit registers

**Review by toastbucket — CHANGES_REQUESTED** [review]

File: src/cpu/test.rs [hunk 5]
```diff
@@ -1508,6 +1508,41 @@ fn test_ld_sp_hl() {
...
 }
 
+// Verify loading contents of SP + s8 to HL
+#[test]
+fn test_ld_hl_sp_plus_s8() {
+    let mut cpu = Cpu::new(Rc::new(RefCell::new(Mmu::new())));
+    const INSTRUCTIONS_LEN: usize = 6;
+    let test_ram: [u8; INSTRUCTIONS_LEN] = [
+        Instruction::LdSpPlusS8ToHl().as_byte(),
+        0xff,
```
Comment (toastbucket) [inline src/cpu/test.rs:1518]: trailing whitespace

File: src/cpu/test.rs [hunk 6]
```diff
@@ -1508,6 +1508,41 @@ fn test_ld_sp_hl() {
...
+        0xff, 
+        Instruction::LdSpPlusS8ToHl().as_byte(),
+        0x1,
+        Instruction::LdSpPlusS8ToHl().as_byte(),
+        0x8,
+    ];
+
+    cpu.load_test_ram(&test_ram);
+
+    // if SP = 0 and s8 = -1 (0xff), ADDING that value will NOT result in carry
```
Comment (toastbucket) [inline src/cpu/test.rs:1527]: set SP here so its clear

File: src/cpu/test.rs [hunk 7]
```diff
@@ -1508,6 +1508,41 @@ fn test_ld_sp_hl() {
     assert_eq!(cpu.get_reg_16(Register16Bit::SP), 0x69);
 }
 
+// Verify loading contents of SP + s8 to HL
+#[test]
+fn test_ld_hl_sp_plus_s8() {
```
Comment (toastbucket) [inline src/cpu/test.rs:1513]: Also test adding -1 to 0xffff (this should result in 0xfffe and both carry and half carry being set)

File: src/cpu/mod.rs [hunk 8]
```diff
@@ -841,6 +841,18 @@ impl Cpu {
                 self.set_reg_16(Register16Bit::SP, hl);
                 (pc + 1, 2)
             },
+            Instruction::LdSpPlusS8ToHl() => {
```
Comment (toastbucket) [inline src/cpu/mod.rs:844]: We're now copying boilerplate multiple times.  Add a TODO to consolidate ALU operations .

Comment (toastbucket) [reply src/cpu/mod.rs:844]: Also put this in its own function

**Review by toastbucket — CHANGES_REQUESTED** [review]

File: src/cpu/instruction.rs [hunk 9]
```diff
@@ -73,6 +73,7 @@ pub enum Instruction {
     LdToCMem(),
     LdFromCMem(),
     LdHlToSp(), 
+    LdSpPlusS8ToHl(), // honestly out of name ideas
```
[the remark this answers is not in the document]
Comment (ricfehr3) [reply src/cpu/instruction.rs:76]: okie dokie

File: src/cpu/test.rs (hunk 5 above)
Comment (ricfehr3) [reply src/cpu/test.rs:1517]: no

Comment (toastbucket) [reply src/cpu/test.rs:1517]: there is though lol

File: src/cpu/mod.rs (hunk 8 above)
Comment (toastbucket) [reply src/cpu/mod.rs]: Still no TODO comment, please add dis

Comment (ricfehr3) [reply src/cpu/mod.rs]: oops sorry, imma add

File: src/cpu/test.rs (hunk 5 above)
Comment (ricfehr3) [reply src/cpu/test.rs:1517]: shit, one sec

**Comment by ricfehr3** [comment]
@toastbucket All shud b gud now

**Review by toastbucket — APPROVED** [review]

**Pull request merged**